### DISCUSSION TITLE
camera model

### DISCUSSION THREAD

**Discussion by flyaudio** [ifm/ifm3d#372] [Q&A]
I want to know more about the distortion model.   

I am going to  use the amplitude image to get the relative pose between chess-board and camera, so I refer to the example "Image_Rectification.ipynb", I found that the link "http://www.vision.caltech.edu/bouguetj/calib_doc/htmls/parameters.html" is out of reach. Is there any other documents explaning the distortion model or the rectification method ?

following the example "Image_Rectification.ipynb", I wonder what 's the intrinsic(fx, fy, cx, cy, especially cx, cy) after rectification ? so I can use PnP(a method provide by opencv) to get the pose between chess-board and camera.

**Comment by lola-masson** [comment]
We use multiple models: for the O3Rxxx 2D cameras, use the fish eye model; for the O3R225 3D camera, use the fish eye model as well; for the O3R222 3D camera, use the Bouguet model. For the fish eye, we mostly follow the model described [here](https://docs.opencv.org/4.x/db/d58/group__calib3d__fisheye.html). The Bouguet model (= pinnhole model) is explained for instance [here](https://towardsdatascience.com/camera-calibration-fda5beb373c3).

**Reply by flyaudio** [reply]
what about the O3D303 ?

**Reply by desengph** [reply]
The O3D303 (as well as most older ifm cameras: O3X, ...) use a Bouguet model for calibration: 
i.e. the calibration parameters or the unit vectors are given in the Bouguet model calibration model.

For older cameras, new calibration models will not be released, as they don't require specialized models for wide angles. The Bouguet model, and it's calibration accuracy (reprojection error) are sufficient in our mind.

**Comment by desengph** [comment]
_I am going to use the amplitude image to get the relative pose between chess-board and camera, so I refer to the example "Image_Rectification.ipynb", I found that the link "http://www.vision.caltech.edu/bouguetj/calib_doc/htmls/parameters.html" is out of reach. Is there any other documents explaning the distortion model or the rectification method ?_

Please see the attached links for further information:
https://johnloomis.org/ece564/notes/CameraCalibration/2012_0524/example_calib.html
http://hvrl.ics.keio.ac.jp/charmie/doc/CameraCalibration/BouguetCameraCalibrationToolbox.pdf
